rules-reviewed/eap7/eap6/hsearch.windup.xml

```diff
@@ -0,0 +1,1197 @@
+                        title="Hibernate Search javadoc for Stoppable" />
+                    <link
+                        href="http://docs.jboss.org/hibernate/search/5.0/api/org/hibernate/search/engine/service/spi/Startable.html"
+                        title="Hibernate Search javadoc for Startable" />
+                    <tag>hibernate-search</tag>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="hsearch-00115">
+            <when>
+                <javaclass references="org.hibernate.search.annotations.IndexedEmbedded">
+                    <location>ANNOTATION</location>
+                </javaclass>
+            </when>
+            <perform>
+                <hint title="Indexing of id Fields of Embedded Relations" effort="1" severity="mandatory">
+                    <message><![CDATA[
+                     When using an `@IndexedEmbedded` annotation to include fields from a related entity,
+                     the id of the related entity is no longer included.
+
+                     You can enable the inclusion of the id by using the `includeEmbeddedObjectId` attribute of the `@IndexedEmbedded` annotation.
+
+                     ```java
+                     @IndexedEmbedded(includeEmbeddedObjectId=true)
+                     ```
+
+                    ]]></message>
+                    <link
+                        href="https://access.qa.redhat.com/documentation/en/red-hat-jboss-enterprise-application-platform/version-7.0/migration-guide/#migrate_miscellaneous_hibernate_search_changes"
+                        title="Miscelanous Changes in Hibernate Search 5.x" />
+                    <link
+                        href="https://docs.jboss.org/hibernate/search/5.5/api/index.html?org/hibernate/search/annotations/IndexedEmbedded.html"
+                        title="Hibernate Search javadoc for IndexedEmbedded" />
+                    <tag>hibernate-search</tag>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="hsearch-00116">
+            <when>
+                <or>
+                    <javaclass references="org.hibernate.search.annotations.Field">
+                        <location>ANNOTATION</location>
+                        <annotation-literal name="index" pattern="Index.YES"/>
+                        <annotation-type pattern="org.hibernate.search.annotations.NumericFields"/>
+                    </javaclass>
+                    <javaclass references="org.hibernate.search.annotations.Field">
+                        <location>ANNOTATION</location>
+                        <annotation-literal name="index" pattern="Index.YES"/>
+                        <annotation-type pattern="org.hibernate.search.annotations.NumericField"/>
+                    </javaclass>
+                </or>
+            </when>
+            <perform>
+                <hint title="Hibernate Search 5 changes indexing null values" effort="1" severity="mandatory">
```

Comment:
Also this rule doesn't catch `Date` and `Calendar` fields annotated with `@DateBridge`/`@CalendarBridge`.